In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('nobel_final.csv')
df.head() #взгляем на первые 5 строк таблицы, чтобы иметь предствавление о данных

firstname    surname born_country_code died_country_code gender  year  \
0  Wilhelm Conrad    Röntgen                DE                DE   male  1901   
1      Hendrik A.    Lorentz                NL                NL   male  1902   
2          Pieter     Zeeman                NL                NL   male  1902   
3           Henri  Becquerel                FR                FR   male  1903   
4          Pierre      Curie                FR                FR   male  1903   

  category  share                                 name_of_university  \
0  physics      1                                  Munich University   
1  physics      2                                  Leiden University   
2  physics      2                               Amsterdam University   
3  physics      2                                École Polytechnique   
4  physics      4  École municipale de physique et de chimie indu...   

  city_of_university country_of_university born_month  age  age_get_prize  
0             Munich               Germany        Mar   78             56  
1             Leiden       the Netherlands        Jul   75             49  
2          Amsterdam       the Netherlands        May   78             37  
3              Paris                France        Dec   56             51  
4              Paris                France        May   47             44

In [3]:
df.rename(columns = {'firstname': 'Имя', 'surname': 'Фамилия', 'born_country_code': 'Код страны, где родился', 'died_country_code': 'Код страны, где умер', 'gender': 'Пол', 'year': 'Год вручения премии', 'category': 'Область науки', 'share': 'Доля/часть премии', 'name_of_university': 'Название университета', 'city_of_university': 'Город университета', 'country_of_university': 'Страна университета', 'born_month': 'Месяц рождения', 'age': 'Сколько лет прожил', 'age_get_prize': 'Возраст при вручении премии'}, inplace = True)
# для своего удобства поменяю названия столбцов (переведу на русский язык), применю изменения к исходному df

In [4]:
df.tail()

Имя    Фамилия Код страны, где родился Код страны, где умер     Пол  \
918    Peter     Handke                      AT                  NaN    male   
919     Abiy  Ahmed Ali                      ET                  NaN    male   
920  Abhijit   Banerjee                      IN                  NaN    male   
921   Esther      Duflo                      FR                  NaN  female   
922  Michael     Kremer                      US                  NaN    male   

     Год вручения премии Область науки  Доля/часть премии  \
918                 2019    literature                  1   
919                 2019         peace                  1   
920                 2019     economics                  3   
921                 2019     economics                  3   
922                 2019     economics                  3   

                           Название университета Город университета  \
918                                          NaN                NaN   
919                                          NaN                NaN   
920  Massachusetts Institute of Technology (MIT)       Cambridge MA   
921  Massachusetts Institute of Technology (MIT)       Cambridge MA   
922                           Harvard University       Cambridge MA   

    Страна университета Месяц рождения  Сколько лет прожил  \
918                 NaN            Dec                  78   
919                 NaN            Aug                  44   
920                 USA            Feb                  59   
921                 USA            Oct                  48   
922                 USA            Nov                  56   

     Возраст при вручении премии  
918                           77  
919                           43  
920                           58  
921                           47  
922                           55

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 923 entries, 0 to 922
Data columns (total 14 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   Имя                          923 non-null    object
 1   Фамилия                      921 non-null    object
 2   Код страны, где родился      923 non-null    object
 3   Код страны, где умер         617 non-null    object
 4   Пол                          923 non-null    object
 5   Год вручения премии          923 non-null    int64 
 6   Область науки                923 non-null    object
 7   Доля/часть премии            923 non-null    int64 
 8   Название университета        697 non-null    object
 9   Город университета           692 non-null    object
 10  Страна университета          693 non-null    object
 11  Месяц рождения               923 non-null    object
 12  Сколько лет прожил           923 non-null    int64 
 13  Возраст при вручении премии  923 no

1.	Сколько лет было самому молодому лауреату, когда он получил премию?

In [6]:
min_age_get_prize = df['Возраст при вручении премии'].min()
min_age_get_prize

17

*Ответ:* самому молодому лауреату было 17 лет на момент вручения премии.

2.	Сколько лет было самому пожилому лауреату, когда он получил премию?

In [7]:
max_age_get_prize = df['Возраст при вручении премии'].max()
max_age_get_prize

97

*Ответ:* самому пожилому лауреату было 97 лет на момент вручения премии.

3.	Сколько всего премий было вручено за период 1901-2019?

In [8]:
df.shape[0] #количество строк в таблице сообщает нам о количестве премий

923

*Ответ:* всего было вручено 923 премии за указанный период.

4.	В каких признаках есть пропущенные значения?

In [9]:
df.isna().sum() #узнаем сумму пропущенных значений по столбцам

Имя                              0
Фамилия                          2
Код страны, где родился          0
Код страны, где умер           306
Пол                              0
Год вручения премии              0
Область науки                    0
Доля/часть премии                0
Название университета          226
Город университета             231
Страна университета            230
Месяц рождения                   0
Сколько лет прожил               0
Возраст при вручении премии      0
dtype: int64

*Ответ:* в столбцах "фамилия", "код страны, где умер (лауреат)", "название университета", "город университета", "страна университета" есть пропущенные значения.

5.	Найдите средний возраст, в котором получали премию.

In [10]:
average_age_get_prize = df['Возраст при вручении премии'].mean()
print(round(average_age_get_prize, 3)) #округлим до трех знаков после запятой

59.823


*Ответ:* средний возраст получения премии равен 59,8 лет.

6.	Выполните описательную статистику по количественным признакам. Сделайте выводы.

In [11]:
df.describe()

Год вручения премии  Доля/часть премии  Сколько лет прожил  \
count           923.000000         923.000000          923.000000   
mean           1971.102925           2.020585           79.346696   
std              33.606425           0.943415           11.134606   
min            1901.000000           1.000000           23.000000   
25%            1947.000000           1.000000           73.000000   
50%            1977.000000           2.000000           80.000000   
75%            2000.000000           3.000000           87.500000   
max            2019.000000           4.000000          103.000000   

       Возраст при вручении премии  
count                   923.000000  
mean                     59.823402  
std                      12.571109  
min                      17.000000  
25%                      51.000000  
50%                      60.000000  
75%                      69.000000  
max                      97.000000

*Вывод:* поскольку представленные выше данные - количественные, то мы можем проанализировать число непропущенных значений, среднее, стандартное отклонение, медиану, 0.25 и 0.75 квартили, максимальное и минимальное значения по каждому столбцу. К примеру, самый ранний год вручения премии - 1901, самый последний - 2019. 75% лауреатов получили свои нобелевские премии до 2000 года. Средняя доля премии, выдаваемой лауреатам, составляет около 2.02 - это значит, что в среднем каждый лауреат делится премией с ещё одним человеком, минимальное значение 1 - одна доля на одного человека, то есть индивидуальная премия, максимум - 4, то есть премия будет разделена на четверых. Меньше всего прожил лауреат, который получил свою премию в 17 и умер в 23 года, а самому пожилому было 103 года, когда он скончался. Самому молодому получателю премии было 17 лет, а самому пожилому - 97 лет. 25% лауреатов получили свои премии до того, как им исполнился 51 год, остальные получали премию уже в более позднем возрасте. По всем 4-м столбцам пропущенные значения отсутсвуют, так как count для всех столбцов равен 923.

7.	Выполните описательную статистику по качественным признакам. Сделайте выводы.

In [ ]:
df.describe(include=['object'])

Имя Фамилия Код страны, где родился Код страны, где умер   Пол  \
count    923     921                     923                  617   923   
unique   704     866                      79                   46     2   
top     John   Smith                      US                   US  male   
freq      12       5                     274                  224   869   

       Область науки     Название университета Город университета  \
count            923                       697                692   
unique             6                       307                186   
top         medicine  University of California       Cambridge MA   
freq             219                        32                 50   

       Страна университета Месяц рождения  
count                  693            923  
unique                  29             12  
top                    USA            Jun  
freq                   358             88

*Вывод:* поскольку представленные выше данные - качественные, то выполняя описательную статистику мы получаем количество непропущенных значений, уникальные значения, самые популярные значения и их частоту по каждому столбцу (таких столбцов здесь 10). Таким образом самое популярное имя среди лауреатов это John, оно встречается 12 раз во всей таблице, а самая популярная фамилия Smith, она встречается 5 раз, при этом в этом столбце ("Фамилия") есть два пропущенных значения, по всей видимости информацию о фамилиях двух лауретов получить не удалось. Больше всего лауреатов родились в США и умерли там же. В столбце "Код страны, где умер (лауреат)" больше всего пропущенных значений. По столбцу "Пол" можно установить, что чаще премию получали мужчины, чем женщины, получается, что из 923 человек 869 были мужчины. Не удивительно, что уникальных значений всего 2 - женский и мужской пол. Всего было 6 областей науки, в которых ученые могли получить премию и самой популярной областью оказалось медицина. University of California стал самым популярном университетом среди мест работы лауретов, в таблице представлено 307 различных университетов. Cambridge MA занимает первое место по популярности среди всех городов, в которых расположены университеты, он встречается 50 раз. Неудивительно, что США лидирует среди 29 стран, представляющих свои университеты с работающими в них выдающимися учеными-лауреатами. По столбцу "Месяц рождения" можно сделать вывод, что больше всего лауретов родилось в июне - 88 человек.

8.	Примените метод  describe() к признаку «Область науки, в которой вручались премии». Сделайте выводы.

In [12]:
df['Область науки'].describe()

count          923
unique           6
top       medicine
freq           219
Name: Область науки, dtype: object

*Ответ:* всего записей об областях науки 923 (= количеству строк), уникальных названий 6 (medicine, physics, chemistry, literature, peace, economics), чаще всего встречаются записи о премиях в области медицины, таких записей 219.

9.	 Примените метод describe()к признаку по Вашему выбору. Сделайте выводы.

In [13]:
df['Месяц рождения'].describe()

count     923
unique     12
top       Jun
freq       88
Name: Месяц рождения, dtype: object

*Ответ:* Чаще всего встречается месяц Июнь, всего 923 записи, содержащих информацию о месяце рождения, уникальных значений 12, так как месяцев в году с разными названиями 12, месяц Июнь встречается 88 раз в столбце "Месяц рождения".

10. Примените метод  value_counts()  к признаку «Область науки, в которой вручались премии». Сделайте выводы.

In [14]:
df['Область науки'].value_counts()

medicine      219
physics       213
chemistry     184
literature    116
peace         107
economics      84
Name: Область науки, dtype: int64

*Ответ:* в области медицины было вручено 219 премий за период, в области физики - 213 премий, в области химии - 184 премии, в области литературы - 116, в области установления/укрепления мира - 107 премий, в области экономики - 84 премии. Получается, чаще всего вручались премии в области медицины, поскольку их больше всего (219 премий).

11. Сколько было лауреатов, родившихся в мае И получивших премию в 2018 году?

In [15]:
may_birth_2018_prize = df[(df['Месяц рождения'] == 'May') & (df['Год вручения премии'] == 2018)]
may_birth_2018_prize.shape[0]

2

*Ответ:* было всего 2 лауреата, родившихся в мае и получивших премию в 2018 году.

12. Сколько было лауреатов, родившихся в марте ИЛИ получивших премию в области литературы?

In [16]:
# по сути можно суммировать строки, в которых по столбцу "месяц рождения" стоит март, а по столбцу "область науки" - литература
march_birth = df[(df['Месяц рождения'] == 'Mar')]
lit_prize = df[(df['Область науки'] == 'literature')]
d = march_birth.shape[0] + lit_prize.shape[0]
print(d)
# в этом случае учитывается совпадение марта и литературы

191


In [17]:
# или же так:
march_birth_lit_prize = df[(df['Месяц рождения'] == 'Mar') | (df['Область науки'] == 'literature')]
march_birth_lit_prize.shape[0]
# в этом случае исключено совпадение двух условий (март и литература)

183

In [18]:
march_birth_lit_prize = df[(df['Месяц рождения'] == 'Mar') & (df['Область науки'] == 'literature')]
march_birth_lit_prize.shape[0]
# 8 случаев совпадения двух признаков (191 - 183 = 8 - логично)

8

In [19]:
march_birth_lit_prize = df[(df['Месяц рождения'] == 'Mar') & (df['Область науки'] == 'literature')]
march_birth_lit_prize
# вот они - эти 8 строчек

Имя         Фамилия Код страны, где родился Код страны, где умер  \
546    Sully       Prudhomme                      FR                   FR   
556     Paul           Heyse                      DE                   DE   
581    Roger  Martin du Gard                      FR                   FR   
603  Giorgos         Seferis                      TR                   GR   
624  Gabriel  García Márquez                      CO                   MX   
632  Octavio             Paz                      MX                   MX   
639    Dario              Fo                      IT                   IT   
812    Mario    Vargas Llosa                      PE                  NaN   

      Пол  Год вручения премии Область науки  Доля/часть премии  \
546  male                 1901    literature                  1   
556  male                 1910    literature                  1   
581  male                 1937    literature                  1   
603  male                 1963    literature                  1   
624  male                 1982    literature                  1   
632  male                 1990    literature                  1   
639  male                 1997    literature                  1   
812  male                 2010    literature                  1   

    Название университета Город университета Страна университета  \
546                   NaN                NaN                 NaN   
556                   NaN                NaN                 NaN   
581                   NaN                NaN                 NaN   
603                   NaN                NaN                 NaN   
624                   NaN                NaN                 NaN   
632                   NaN                NaN                 NaN   
639                   NaN                NaN                 NaN   
812                   NaN                NaN                 NaN   

    Месяц рождения  Сколько лет прожил  Возраст при вручении премии  
546            Mar                  68                           62  
556            Mar                  84                           80  
581            Mar                  77                           56  
603            Mar                  71                           63  
624            Mar                  87                           55  
632            Mar                  84                           76  
639            Mar                  90                           71  
812            Mar                  84                           74

*Ответ:* таким образом количество лауреатов, родившихся в марте или получивших премию в области литературы, равно 191 (буду учитывать случаи совпадения тоже).